aws-iot-mitigationaction/src/main/java/com/amazonaws/iot/mitigationaction/CreateHandler.java

```diff
@@ -0,0 +1,100 @@
+package com.amazonaws.iot.mitigationaction;
+
+import org.apache.commons.lang3.StringUtils;
+import software.amazon.awssdk.services.iot.IotClient;
+import software.amazon.awssdk.services.iot.model.CreateMitigationActionRequest;
+import software.amazon.awssdk.services.iot.model.CreateMitigationActionResponse;
+import software.amazon.awssdk.services.iot.model.MitigationActionParams;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.resource.IdentifierUtils;
+
+import java.util.Map;
+
+public class CreateHandler extends BaseHandler<CallbackContext> {
+
+    // Copied value from software.amazon.cloudformation.resource.IdentifierUtils
+    private static final int GENERATED_NAME_MAX_LENGTH = 40;
+
+    private final IotClient iotClient;
+
+    public CreateHandler() {
+        iotClient = IotClient.builder().build();
+    }
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            AmazonWebServicesClientProxy proxy,
+            ResourceHandlerRequest<ResourceModel> request,
+            CallbackContext callbackContext,
+            Logger logger) {
+
+        CreateMitigationActionRequest createRequest = translateToCreateRequest(request);
+
+        ResourceModel model = request.getDesiredResourceState();
+        if (!StringUtils.isEmpty(model.getMitigationActionArn())) {
+            logger.log(String.format("MitigationActionArn is read-only, but the caller passed %s.",
+                    model.getMitigationActionArn()));
+            // Note: this is necessary even though MitigationActionArn is marked readOnly in the schema.
+            return ProgressEvent.failed(model, callbackContext, HandlerErrorCode.InvalidRequest,
+                    "MitigationActionArn is a read-only property and cannot be set.");
+        }
+
+        if (!StringUtils.isEmpty(model.getMitigationActionId())) {
+            logger.log(String.format("MitigationActionId is read-only, but the caller passed %s.",
+                    model.getMitigationActionArn()));
```

Comment:
Incorrect field name, should be MitigationActionId.

Reply:
Changed